core/templates/domain/state_card/state-card.model.ts

```diff
@@ -281,7 +281,7 @@ export class StateCard {
   swapContentsWithTranslation(entityTranslation: EntityTranslation): void {
     let writtenTranslation = entityTranslation.getWrittenTranslation(
       this._contentId) as TranslatedContent;
-    if (writtenTranslation && !writtenTranslation.needsUpdate) {
+    if (writtenTranslation) {
```

Comment:
Quick note: A better UX would be to also present a small banner at the bottom of the content with a note that the presented translation is not up-to-date but considering the editor has marked translations needsUpdate and considers that the translated contents can be presented to the learner ([ref](https://github.com/oppia/oppia/blob/a643a222f7ef92efde1c58f6a9ebba90bc860217/core/templates/components/forms/forms-templates/mark-translations-as-needing-update-modal.component.html#L26)) we are not adding any such banner to avoid distracting/confusing learners.